### PR TITLE
Explain how Shamir's Secret Sharing works

### PR DESCRIPTION
**Pull request by autrilla**

### CODE REVIEW

File: shamir/shamir.go.commented [hunk 1]
```diff
@@ -0,0 +1,237 @@
+package shamir
```
Comment (jvehent) [inline shamir/shamir.go.commented]: Why the separate `.commented` file? Can't this be merged directly with shamir.go?

Comment (jvehent) [reply shamir/shamir.go.commented]: @autrilla ^^^

Comment (autrilla) [reply shamir/shamir.go.commented]: Oops, missed this. I'm not sure if we're allowed to modify the shamir file, and I removed some of their comments anyway. We could theoretically merge this, but idk if having an algorithm explanation in the source code alongside implementation details is a lot of noise. Your call.

Comment (jvehent) [reply shamir/shamir.go.commented]: I don't think we should have a separate .commented file. I'd prefer to merge it alongside the source code.

**Comment by jvehent** [comment]
ping @autrilla

**Comment by autrilla** [comment]
done @jvehent

**Pull request merged**